### PR TITLE
Fixed get_infer_data to always return DK cells when infer_labeled=False.

### PR DESCRIPTION
**Pull request by richardwu**
Also renamed parameter names for better consistency and added `estimator_enabled` parameter which disables the weak labelling pathway.

## Changes to `get_infer_data`

When infer_labeled = True, fixing_domain_gen performed inference on ALL cells (clean and DK). This meant that un-detected errors (clean cells that had errors) were also being repaired. This overestimated the # of correct/total repair and thus it is not recommend to set infer_labeled = True when performing experiments.

When infer_labeled = False, fixing_domain_gen only inferred on non-weak labelled cells. If during weak labelling a DK cell was weak labelled, it would not be inferred. This is fixed in https://github.com/HoloClean/holoclean/pull/56 so all DK cells are inferred regardless of their labelling.

### CODE REVIEW

File: holoclean.py [hunk 1]
```diff
@@ -121,6 +121,12 @@
       'default': 0.1,
       'type': float,
       'help': 'Correlation threshold (absolute) when selecting correlated attributes for domain pruning.'}),
+    (('-wlt', '--estimator-enabled'),
+     {'metavar': 'ESTIMATOR_ENABLED',
+      'dest': 'estimator_enabled',
+      'default': True,
+      'type': bool,
+      'help': 'Enables the posterior estimator for weak labelling and domain generation.'}),
```
Comment (minafarid) [inline holoclean.py:133]: We should add either here or up in the domain generation when we skip if this parameter is false (better in my opinion) that we only use the top co-occurring values as the cell domain (if that is a valid statement)

File: holoclean.py [hunk 2]
```diff
@@ -207,6 +213,17 @@ def __init__(self, env, name="session"):
         :param env: Holoclean environment
         :param name: Name for the Holoclean session
         """
+        # use DEBUG logging level if verbose enabled
+        root_logger = logging.getLogger()
+        gensim_logger = logging.getLogger('gensim')
+        root_level, gensim_level = logging.INFO, logging.WARNING
+        if env['verbose']:
```
Comment (minafarid) [inline holoclean.py:221]: keep this if statement here, but move everything else to line 12 with the logging details. I understand we need `env` here, so we only modify the logging level if `verbose`

**Review by minafarid — APPROVED** [review]
LGTM with minor comments.

File: holoclean.py (hunk 1 above)
Comment (richardwu) [reply holoclean.py:133]: I'm going to do a large refactor of naming of the parameters and what's surfaced after I finalize the experiments. Going to merge this into `dev` for consistency.

**Pull request merged**